pkg/flowaggregator/flowaggregator_test.go

```diff
@@ -278,3 +287,82 @@ func createElement(name string, enterpriseID uint32) ipfixentities.InfoElementWi
 	ieWithValue, _ := ipfixentities.DecodeAndCreateInfoElementWithValue(element, nil)
 	return ieWithValue
 }
+
+func TestFlowAggregator_watchConfiguration(t *testing.T) {
+	dbEP := &clickhouseclient.ClickHouseExportProcess{}
+	var wg sync.WaitGroup
+	testcases := []struct {
+		FlowCollectorEnable       bool
+		FlowCollectorAddress      string
+		ClickHouseEnable          bool
+		externalFlowCollectorAddr string
+		message                   []updateParam
+		dbExportProcess           *clickhouseclient.ClickHouseExportProcess
+	}{
+		{true, "10.10.10.10:155", false, "", []updateParam{updateExternalFlowCollectorAddr}, nil},
+		{false, "", true, "addr", []updateParam{disableFlowCollector}, nil},
+		{false, "", true, "", []updateParam{enableClickHouse}, nil},
+		{false, "", true, "", []updateParam{updateClickHouseParam}, dbEP},
+		{true, "10.10.10.10:155", false, "", []updateParam{updateExternalFlowCollectorAddr, disableClickHouse}, dbEP},
+	}
+
+	for i, tc := range testcases {
+		t.Run(fmt.Sprintf("subtest: %d", i), func(t *testing.T) {
+			stopCh := make(chan struct{})
+			opt := options.Options{
+				Config: &flowaggregatorconfig.FlowAggregatorConfig{
+					FlowCollector: flowaggregatorconfig.FlowCollectorConfig{
+						Enable:  tc.FlowCollectorEnable,
+						Address: tc.FlowCollectorAddress,
+					},
+					ClickHouse: flowaggregatorconfig.ClickHouseConfig{
+						Enable: tc.ClickHouseEnable,
+					},
+				},
+			}
+			// create watcher
+			fileName := fmt.Sprintf("test_%d.config", i)
+			configWatcher, err := fsnotify.NewWatcher()
+			assert.NoError(t, err)
+			flowAggregator := &flowAggregator{
+				// use a larger buffer to prevent the buffered channel from blocking
+				updateCh:                  make(chan updateMsg, 100),
+				externalFlowCollectorAddr: tc.externalFlowCollectorAddr,
+				dbExportProcess:           tc.dbExportProcess,
+				configFile:                fileName,
+				configWatcher:             configWatcher,
+			}
+			dir := filepath.Dir(fileName)
+			f, err := os.Create(flowAggregator.configFile)
+			assert.NoError(t, err)
+			err = flowAggregator.configWatcher.Add(dir)
+			assert.NoError(t, err)
+			err = f.Close()
+			assert.NoError(t, err)
+			err = os.Remove(flowAggregator.configFile)
+			assert.NoError(t, err)
+			f, err = os.Create(flowAggregator.configFile)
+			assert.NoError(t, err)
+			b, err := yaml.Marshal(opt.Config)
+			assert.NoError(t, err)
+			_, err = f.Write(b)
+			assert.NoError(t, err)
+			err = f.Close()
+			assert.NoError(t, err)
+			wg.Add(1)
+			go flowAggregator.watchConfiguration(stopCh, &wg)
+
+			for _, message := range tc.message {
+				select {
+				case msg := <-flowAggregator.updateCh:
+					assert.Equal(t, message, msg.param)
+				case <-time.After(1 * time.Minute):
+					assert.NoError(t, fmt.Errorf("timeout"))
+				}
+			}
+			close(stopCh)
+			wg.Wait()
```

Comment:
It's a bad idea to add a WaitGroup to the parameter list of `watchConfiguration` just for the sake of testing. And I still don't understand why it is required.
It feels like this code could be:
```go
wg.Add(1)
go func() {
        defer wg.Done()
        flowAggregator.watchConfiguration(stopCh)
}()
...
close(stopCh)
wg.Wait()
```

Reply:
mb, I didn't figure out this easiest way to use wg and channel.
I've fixed it. Thanks for the suggestion.